server/common/rest.py

```diff
@@ -0,0 +1,205 @@
+        if annotations is not None:
+            labels = annotations.read_labels(data_engine)
+        return make_response(
+            data_engine.annotation_to_fbs_matrix(Axis.VAR, fields, labels),
+            HTTPStatus.OK,
+            {"Content-Type": "application/octet-stream"},
+        )
+    except KeyError:
+        return make_response(f"Error bad key in {fields}", HTTPStatus.BAD_REQUEST)
+    except ValueError as e:
+        return make_response(str(e), HTTPStatus.INTERNAL_SERVER_ERROR)
+    except Exception as e:
+        return make_response(str(e), HTTPStatus.INTERNAL_SERVER_ERROR)
+
+
+def data_var_put(request, data_engine):
+    preferred_mimetype = request.accept_mimetypes.best_match(["application/octet-stream"])
+    if preferred_mimetype != "application/octet-stream":
+        return make_response(f"Unsupported MIME type '{request.accept_mimetypes}'", HTTPStatus.NOT_ACCEPTABLE)
+
+    filter_json = request.get_json()
+    filter = filter_json["filter"] if filter_json else None
+    try:
+        return make_response(
+            data_engine.data_frame_to_fbs_matrix(filter, axis=Axis.VAR),
+            HTTPStatus.OK,
+            {"Content-Type": "application/octet-stream"},
+        )
+    except FilterError as e:
+        return make_response(str(e), HTTPStatus.BAD_REQUEST)
+    except ValueError as e:
+        return make_response(str(e), HTTPStatus.INTERNAL_SERVER_ERROR)
+
+
+def diffexp_obs_post(request, data_engine):
+    args = request.get_json()
+    # confirm mode is present and legal
+    try:
+        mode = DiffExpMode(args["mode"])
+    except KeyError:
+        return make_response("Error: mode is required", HTTPStatus.BAD_REQUEST)
+    except ValueError:
+        return make_response(f"Error: invalid mode option {args['mode']}", HTTPStatus.BAD_REQUEST)
+    # Validate filters
+    if mode == DiffExpMode.VAR_FILTER or "varFilter" in args:
+        # not NOT_IMPLEMENTED
+        return make_response("mode=varfilter not implemented", HTTPStatus.NOT_IMPLEMENTED)
+    if mode == DiffExpMode.TOP_N and "count" not in args:
+        return make_response("mode=topN requires a count parameter", HTTPStatus.BAD_REQUEST)
+
+    if "set1" not in args:
+        return make_response("set1 is required.", HTTPStatus.BAD_REQUEST)
+    if Axis.VAR in args["set1"]["filter"]:
+        return make_response("Var filter not allowed for set1", HTTPStatus.BAD_REQUEST)
+    # set2
+    if "set2" not in args:
+        return make_response("Set2 as inverse of set1 is not implemented", HTTPStatus.NOT_IMPLEMENTED)
+    if Axis.VAR in args["set2"]["filter"]:
+        return make_response("Var filter not allowed for set2", HTTPStatus.BAD_REQUEST)
+
+    set1_filter = args["set1"]["filter"]
+    set2_filter = args.get("set2", {"filter": {}})["filter"]
+
+    # TODO: implement varfilter mode
+
+    # mode=topN
+    count = args.get("count", None)
+    try:
+        interactive_limit = data_engine.get_features()["diffexp"].interactiveLimit
```

Comment:
interactive_limits were never implemented properly, and the front-end does not understand them.  Suggest removing this dead code.

Reply:
Done